tests/backup/backup_helper.go

```diff
@@ -1225,6 +1280,48 @@ func restoreSuccessCheck(restoreName string, orgID string, retryDuration time.Du
 	return nil
 }
 
+// restoreSuccessWithReplacePolicy inspects restore task status as per ReplacePolicy_Type
+func restoreSuccessWithReplacePolicy(restoreName string, orgID string, retryDuration time.Duration, retryInterval time.Duration, ctx context.Context, replacePolicy ReplacePolicy_Type) error {
+	restoreInspectRequest := &api.RestoreInspectRequest{
+		Name:  restoreName,
+		OrgId: orgID,
+	}
+	var statusesExpected api.RestoreInfo_StatusInfo_Status
+	if replacePolicy == ReplacePolicy_Delete {
+		statusesExpected = api.RestoreInfo_StatusInfo_Success
+	} else if replacePolicy == ReplacePolicy_Retain {
+		statusesExpected = api.RestoreInfo_StatusInfo_PartialSuccess
+	}
+	statusesUnexpected := [...]api.RestoreInfo_StatusInfo_Status{
+		api.RestoreInfo_StatusInfo_Invalid,
+		api.RestoreInfo_StatusInfo_Aborted,
+		api.RestoreInfo_StatusInfo_Failed,
+	}
+	restoreSuccessCheckFunc := func() (interface{}, bool, error) {
+		resp, err := Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
+		if err != nil {
+			return "", false, err
+		}
+		actual := resp.GetRestore().GetStatus().Status
+		reason := resp.GetRestore().GetStatus().Reason
+		if actual == statusesExpected {
+			return "", false, nil
+		}
+
+		for _, status := range statusesUnexpected {
+			if actual == status {
+				return "", false, fmt.Errorf("restore status for [%s] expected was [%v] but got [%s] because of [%s]", restoreName, statusesExpected, actual, reason)
+			}
+		}
+		return "", true, fmt.Errorf("restore status for [%s] expected was [%v] but got [%s] because of [%s]", restoreName, statusesExpected, actual, reason)
+	}
+	_, err := task.DoRetryWithTimeout(restoreSuccessCheckFunc, retryDuration, retryInterval)
+	if err != nil {
+		return err
+	}
```

Comment:
just do return err here, it can be error or nil

Reply:
done